mpcforces_extractor/datastructure/rigids.py

```diff
@@ -34,20 +36,24 @@ def __init__(
         self.mpc_config: MPC_CONFIG = mpc_config
         if master_node is None:
             Logger().log_warn("Master_node2coords is None for element_id", element_id)
-        self.master_node = master_node
+        self.master_node: Node = master_node
         self.nodes: List = nodes
         self.dofs: int = dofs
         self.part_id2node_ids = {}
+        self.axis = None
+        self.diameter = None
+        self.length = None
+        self.area = None
 
         # config_2_id_2_instance
         if mpc_config.value not in MPC.config_2_id_2_instance:
             MPC.config_2_id_2_instance[mpc_config.value] = {}
-
         if element_id in MPC.config_2_id_2_instance[mpc_config.value]:
             Logger().log_err(
                 f"Element with id {element_id} already exists for config {mpc_config}"
             )
         MPC.config_2_id_2_instance[mpc_config.value][element_id] = self
+        MPC.all_instances.append(self)
```

Comment:
this has to have implications on the fronend - right? Should be followed through

Reply:
Surprisingly, it doesn't  - it makes sense to use the config_2_id_2_instance variable in the populate_database method and not to use the all_instances.